solar_system.rb

```diff
@@ -0,0 +1,211 @@
+      continue = gets.chomp.upcase
+      continue = y_or_n(continue)
+    end
+  else
+    puts "Ok, I won't tell you any more about my planets"
+  end
+end
+
+#prompts the user to input all the the information to define a new planet
+#and makes the new planet
+def make_a_planet(name)
+  puts "How long is a year on #{name}? (in earth-days)"
+  year_length = gets.chomp
+  puts "How far form the sun is #{name} (in units)?"
+  distance_from_sun = gets.chomp
+  puts "If this was a star trek episode, what type of planet is #{name}?"
+  type = gets.chomp
+  puts "How many moons does #{name} have?"
+  moons = gets.chomp
+  puts "Does an intelligent species live on #{name}? (Y/N)"
+  species = gets.chomp.upcase
+  species = y_or_n(species)
+  if species == "N"
+    inhabitants = "No inteligent life"
+  else
+    puts "What is the species name?"
+    inhabitants = gets.chomp
+  end
+  new_planet = Planet.new(name, year_length, distance_from_sun, type, moons, inhabitants)
+  return new_planet
+end
+
+#asks the user if they would like to define a new planet.
+# If yes, it allows them too.
+def user_defined_planet(my_solar_system)
+  puts "I can help you define a new planet. Would you like to?"
+  wants_to_define = gets.chomp.upcase
+  wants_to_define = y_or_n(wants_to_define)
+
+  if wants_to_define == "Y"
+    puts "Great! Please enter the name of your new planet"
+    name = gets.chomp.capitalize
+    new_planet = make_a_planet(name)
+    puts "This is the information on the planet you just entered"
+    puts new_planet.planet_all_info
+    #could ask user if that info is correct here, and allow changes if not.
+    my_solar_system.add_planet(new_planet)
+    puts "\nHere is the updated list of planets in my solar system"
+    puts my_solar_system.ss_name_list
+    puts ""
+  else
+    puts "Well OK then."
+  end
+  return my_solar_system
+end
+
+#SETTING VARIABLES
+#these are hardcoded planets in my starter solar system
+first = Planet.new("Caprica", 102, 230, "M", 4, "Humans")
+second = Planet.new("Vulcan", 40, 100 ,"M", 2, "Vulcans")
+third = Planet.new("Delta Vega", 50, 1100, "A", 1, "No inteligent life" )
+forth = Planet.new("Alpha Onias III", 102, 230, "M", 1, "Barash")
+
+#build my starter solar system
+my_solar_system = SolarSystem.new([first, second, third, forth])
+
+puts "\nHere are the planets in my solar system"
+puts planets_in_my_solar_system = my_solar_system.ss_name_list
+puts ""
+
+learn_about_system(my_solar_system)
+my_solar_system =user_defined_planet(my_solar_system)
+#by going back throug the learn_about_system this demonstrates that you
+#can now learn about the planet you just entered.
+learn_about_system(my_solar_system)
+#could have automatic way to cycle through these.  Also include a quit option.
+
+
+
+#This method to determine distance between planets works, but it is very ugly
+# def distance_from_each_other(my_solar_system, planet1_name, planet2_name)
+#   i=0
```

Comment:
A few thoughts on this code:

1. Since this method takes a solar system as an argument, it might be another good candidate for an instance method
1. You've got the code to select a planet by name repeated twice - this could be extracted to its own method
1. In general, when working with arrays `each` loops are easier to read than `times` loops
1. Putting a `$` in front of a variable name make it a global variable, which has all sorts of interesting implications and is almost certainly not what you want
1. The last few lines could be substantially shortened using the `abs` (absolute value) method

An all the way shortened version might read like:

```ruby
class SolarSystem

  # ... whatever is already in this class ...

  def lookup_planet(name)
    @planets.each do |planet|
      if planet.name == name
        return planet
      end
    end
    return nil
  end

  def distance_between(planet1_name, planet2_name)
    planet1 = lookup_planet(planet1_name)
    planet2 = lookup_planet(planet2_name)
    return (planet1.distance_from_sun - planet2.distance_from_sun).abs
  end
end
```

Don't worry if your code isn't quite to this point yet - if it was, I don't know what you'd be doing in this class! Solving the problem is priority 1; making it easy to read will come with time.